nyxtools/flyer.py

```diff
@@ -20,93 +23,183 @@ def __init__(self, vector, zebra, detector=None) -> None:
         self.zebra = zebra
         self.detector = detector
 
+        self.data_directory_name = None
+        self.file_prefix = None
+        self.num_images = None
+        self.file_number_start = None
+
         self._asset_docs_cache = deque()
-        self._resource_uids = []
-        self._datum_counter = None
-        self._datum_ids = DEFAULT_DATUM_DICT
+        self._resource_document = None
+        self._datum_factory = None
+
+        # self._resource_uids = []
+        # self._datum_counter = None
+        # TODO: rework to use datum ids dictionary.
+        # self._datum_ids = DEFAULT_DATUM_DICT
+        self._datum_ids = []
         self._first_file = None
 
         self._collection_dictionary = None
 
+    def update_parameters(self, **kwargs):
+        self.data_directory_name = kwargs.get("data_directory_name", "/nyx-data/test")
+        self.file_prefix = kwargs.get("file_prefix", "test")
+        self.num_images = kwargs.get("num_images", 1)
+        self.file_number_start = kwargs.get("file_number_start", 1)
+
+        super().update_parameters(**kwargs)
+
     def kickoff(self):
         self.detector.stage()
-        self.vector.move()
 
-        return NullStatus()
+        def zebra_callback(*args, **kwargs):
+            print(f"\n{ttime.ctime()}:\n\targs: {args}\n\tkwargs: {kwargs}\n")
```

Comment:
We should replace these with logger.info or logger.debug statements, if we want to leave them in.

Reply:
I can do that, if you guys agree.

Reply:
Yes, please go ahead, @RobertSchaffer1. I'll be happy to review your changes.